sheet2linkml/cli.py

```diff
@@ -49,30 +57,55 @@
     default=True,
     help="Controls whether we use the CCDH Terminology Service to add enumerated values for attributes.",
 )
-def main(output, filter_entity, logging_config, write_mappings, include_terminologies):
+def main(
+    google_sheet_id,
+    output,
+    filter_entity,
+    logging_config,
+    write_mappings,
+    include_terminologies,
+):
     # Display INFO log entry and up.
-    logging.config.fileConfig(logging_config)
+    if os.path.exists(logging_config):
+        logging.config.fileConfig(logging_config)
 
     # Load environemental variables from `.env` if one is present.
     load_dotenv()
 
     # Read in Google API credentials.
-    google_api_credentials = os.getenv(
-        "GOOGLE_API_CREDENTIALS", "google_api_credentials.json"
-    )
-    if not os.path.exists(google_api_credentials):
-        logging.error(
-            f"Google API Credential file '{google_api_credentials}' not found: please see "
-            + "https://github.com/cancerDHC/sheet2linkml#authorization for information on creating this file."
+    if "SHEET2LINKML_GOOGLE_SERVICE_ACCT" in os.environ:
+        # If an environment variable SHEET2LINKML_GOOGLE_SERVICE_ACCT is set, use it as a
+        # Service Account (https://pygsheets.readthedocs.io/en/stable/authorization.html#service-account)
+        pygclient = pygsheets.authorize(
+            service_account_env_var="SHEET2LINKML_GOOGLE_SERVICE_ACCT",
+            scopes=GSheetModel.SCOPES,
+        )
+    else:
+        # Otherwise, look for the path to Google API Credentials in google_api_credentials.json.
+        google_api_credentials = os.getenv(
+            "GOOGLE_API_CREDENTIALS", "google_api_credentials.json"
+        )
+        if not os.path.exists(google_api_credentials):
+            logging.error(
+                f"Google API Credential file '{google_api_credentials}' not found: please see "
+                + "https://github.com/cancerDHC/sheet2linkml#authorization for information on creating this file."
+            )
+            exit(1)
+        pygclient = pygsheets.authorize(
+            client_secret=google_api_credentials, scopes=GSheetModel.SCOPES
         )
-        exit(1)
-    google_sheet_id = os.getenv("CDM_GOOGLE_SHEET_ID")
 
     # Arbitrarily set a CRDC-H root URI.
     crdch_root = "https://example.org/crdch"
 
+    if not google_sheet_id:
+        logging.error(
+            "A Google Sheet ID is required; please set environmental variable 'CDM_GOOGLE_SHEET_ID' to a Google Sheet ID."
+        )
+        exit(1)
```

Comment:
Looking at this [SO answer](https://stackoverflow.com/questions/6501121/difference-between-exit-and-sys-exit-in-python) that describes the difference between `exit()` and `sys.exit()` it's becoming more apparent to me that we need to use `sys.exit()` since this isn't an interactive shell application?

Code:

```
import sys

# python code

sys.exit()
```

Reply:
You are correct! See [the Python docs](https://docs.python.org/3/library/constants.html#exit):
> The site module (which is imported automatically during startup, except if the -S command-line option is given) adds several constants to the built-in namespace. They are useful for the interactive interpreter shell and should not be used in programs.

Fixed in 8ebf86e.